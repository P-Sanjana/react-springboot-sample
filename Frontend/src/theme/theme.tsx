import { Color } from "@material-ui/core";
import { createTheme } from "@material-ui/core/styles";
import GraphikMedium from "../assets/fonts/GraphikMedium.otf";
import GraphikRegular from "../assets/fonts/GraphikRegular.otf";
import GraphikSemiBold from "../assets/fonts/GraphikSemibold.otf";

type ColorPartial = Partial<Color>;

declare module "@material-ui/core/styles/createPalette" {
  // eslint-disable-next-line @typescript-eslint/no-empty-interface
  interface PaletteColor extends ColorPartial {}
}

declare module "@material-ui/core/styles/createTypography" {
  interface Typography {
    inhherit: React.CSSProperties;
    caption2: React.CSSProperties;
  }

  interface TypographyOptions {
    inherit?: React.CSSProperties;
    caption2?: React.CSSProperties;
  }
}

declare module "@material-ui/core/Typography/Typography" {
  interface TypographyPropsVariantOverrides {
    caption2: true;
  }
}

const fontFaceGraphikRegular = {
  fontFamily: "GraphikRegular",
  src: `url(${GraphikRegular}) format("opentype")`,
};

const fontFaceGraphikSemiBold = {
  fontFamily: "GraphikSemiBold",
  src: `url(${GraphikSemiBold}) format("opentype")`,
};

const fontFaceGraphikMedium = {
  fontFamily: "GraphikMedium",
  src: `url(${GraphikMedium}) format("opentype")`,
};

const theme = createTheme({
  spacing: [0, 4, 8, 12, 16, 20, 24, 32],
  palette: {
    primary: {
      main: "#0052FF",
      "100": "#FAFCFF",
      "300": "#CCE3FF",
      "500": "#0052FF",
      "700": "#002EB7",
      "900": "#00177A",
    },
    success: {
      main: "#20B03F",
      "500": "#20B03F",
      "100": "#E9F7EC",
    },
    warning: {
      main: "#FFA74F",
      "500": "#FFA74F",
      "100": "#FFF6ED",
    },
    error: {
      main: "#B71A33",
      "500": "#B71A33",
      "100": "#F3E6EB",
    },
    grey: {
      "50": "#F2F2F7",
      "100": "#E8E8F7",
      "300": "#B4B4CF",
      "500": "#4B4B60",
      "700": "#252545",
      "900": "#0E0E2E",
    },
    text: {
      primary: "#343446",
      secondary: "#7D7D89",
      hint: "#7D7D89",
      disabled: "#B2B2B9",
    },
    secondary: {
      main: "#FFA74F",
      contrastText: "#FFFFFF",
    },
  },
  overrides: {
    MuiCssBaseline: {
      "@global": {
        "@font-face": [
          fontFaceGraphikRegular,
          fontFaceGraphikSemiBold,
          fontFaceGraphikMedium,
        ],
      },
    },
  },
  typography: {
    overline: {
      gammaOverlineFontFamily: "Comic Sans Ms ",
      fontStyle: "normal",
      fontWeight: "normal",
      fontSize: "12px",
      lineHeight: "14px",
      identicalToBoxHeightOr117LetterSpacing: "0.005em",
      betaTextHighEmphasisColor: "#343446",
    },
    h4: {
      fontFamily: "GraphikMedium",
      fontSize: "40px",
      fontWeight: 500,
      letterSpacing: "-1%",
      lineHeight: "54px",
    },
    h6: {
      fontFamily: "GraphikRegular",
      fontStyle: "normal",
      fontWeight: 400,
      fontSize: "24px",
      lineHeight: "34px",
    },
    caption: {
      fontFamily: "GraphikMedium",
      fontStyle: "normal",
      fontWeight: 500,
      fontSize: "14px",
      lineHeight: "16px",
      letterSpacing: "0.01em",
    },
    inherit: {
      fontFamily: "GraphikRegular",
      fontStyle: "normal",
      fontWeight: "normal",
      fontSize: "14px",
      lineHeight: "16px",
      letterSpacing: "0.01em",
    },
    button: {
      fontFamily: "GraphikSemiBold",
      fontStyle: "normal",
      fontWeight: 500,
      fontSize: "14px",
      lineHeight: "42px",
      letterSpacing: "0.01em",
    },
    body2: {
      fontFamily: "GraphikRegular",
      fontStyle: "normal",
      fontWeight: "normal",
      fontSize: "16px",
      lineHeight: "22px",
      letterSpacing: "0.01em",
    },
    subtitle2: {
      fontFamily: "GraphikRegular",
      fontStyle: "normal",
      fontWeight: "normal",
      fontSize: "20px",
      lineHeight: "28px",
      letterSpacing: "0.005em",
    },
    subtitle1: {
      fontFamily: "GraphikSemiBold",
      fontStyle: "normal",
      fontWeight: 500,
      fontSize: "20px",
      lineHeight: "28px",
      letterSpacing: "0.005em",
    },
    body1: {
      fontFamily: "GraphikSemiBold",
      fontStyle: "normal",
      fontWeight: 500,
      fontSize: "16px",
      lineHeight: "22px",
      letterSpacing: "0.01em",
    },
    caption2: {
      fontFamily: "GraphikRegular",
      fontStyle: "normal",
      fontWeight: 400,
      fontSize: "14px",
      lineHeight: "16px",
      letterSpacing: "0.01em",
    },
  },
});

export default theme;
