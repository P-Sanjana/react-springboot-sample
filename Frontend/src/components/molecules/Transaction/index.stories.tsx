import React from "react";
import { Story, Meta } from "@storybook/react";
import Transaction, { TransactionProps } from "./index";
import theme from "../../../theme/theme";
import CheckLine from "../../../assets/icons/check-line.svg";
import { Grid } from "@material-ui/core";

export default {
  title: "Molecules/Transaction",
  component: Transaction,
} as Meta;

const Template: Story<TransactionProps> = (args) => (
  <Grid item xs={4}>
    <Transaction {...args} />
  </Grid>
);

export const CardStats = Template.bind({});
CardStats.args = {
  date: "June 23",
  statusIcon: CheckLine,
  coin: "Bitcoin",
  symbol: "BTC",
  chipLabel: "Sold",
  coinAmount: -0.023451,
  price: +34000.0,
  iconBgColor: theme.palette.success["100"],
};
