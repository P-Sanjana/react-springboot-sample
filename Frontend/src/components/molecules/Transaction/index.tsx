import React from "react";
import {
  Box,
  makeStyles,
  Typography,
  Chip,
  Grid,
  styled,
} from "@material-ui/core";
import theme from "../../../theme/theme";
import localeUtils from "../../../utils/locale-utils";

export type TransactionProps = {
  date: string;
  statusIcon: string;
  coin: string;
  symbol: string;
  chipLabel: string;
  coinAmount: number;
  price: number;
  iconBgColor: string;
};

const Caption2Typography = styled(Typography)(({ theme }) => ({
  ...theme.typography.caption2,
}));

const Transaction: React.FC<TransactionProps> = ({
  date,
  statusIcon,
  coin,
  symbol,
  chipLabel,
  coinAmount,
  price,
  iconBgColor,
}) => {
  const useStyles = makeStyles({
    rowContent: {
      display: "flex",
      flexDirection: "row",
      justifyContent: "flex-start",
      alignItems: "center",
    },
    date: {
      height: theme.spacing(4),
      color: theme.palette.text.primary,
    },
    coin: {
      height: "22px",
      color: theme.palette.text.primary,
      margin: theme.spacing(2, 0),
    },
    statusIcon: {
      width: "44px",
      height: "44px",
      borderRadius: "100px",
      backgroundColor: iconBgColor,
    },
    chip: {
      color: theme.palette.grey["500"],
      backgroundColor: theme.palette.grey["100"],
    },
    coinAmount: {
      color: theme.palette.text.primary,
      paddingBottom: theme.spacing(2),
    },
    price: {
      color: theme.palette.text.hint,
    },
    columnContent: {
      display: "flex",
      flexDirection: "column",
      justifyContent: "flex-start",
    },
  });
  const classes = useStyles();
  return (
    <Grid container spacing={2}>
      <Grid item xs={12}>
        <Box>
          <Caption2Typography
            className={classes.date}
            variant="caption"
            data-testid="date"
          >
            {date}
          </Caption2Typography>
        </Box>
        <Box className={classes.rowContent}>
          <Grid item xs={2} alignItems="center">
            <img
              src={statusIcon}
              className={classes.statusIcon}
              data-testid="status"
            />
          </Grid>
          <Grid item xs={10}>
            <Box>
              <Typography
                variant="body1"
                className={classes.coin}
                data-testid="currency"
              >
                {coin} {symbol}
              </Typography>
              <Chip size="small" label={chipLabel} className={classes.chip} />
            </Box>
          </Grid>
          <Grid item xs={3}>
            <Box
              className={classes.columnContent}
              style={{ alignItems: "flex-end" }}
            >
              <Typography variant="body1" className={classes.coinAmount}>
                {chipLabel === "Sold"
                  ? `-${localeUtils.formatCurr(coinAmount)}`
                  : `+${localeUtils.formatCurr(coinAmount)}`}
                {symbol}
              </Typography>
              <Caption2Typography variant="caption" className={classes.price}>
                {chipLabel === "Sold"
                  ? `+${localeUtils.formatCurr(price)}`
                  : `-${localeUtils.formatCurr(price)}`}
              </Caption2Typography>
            </Box>
          </Grid>
        </Box>
      </Grid>
    </Grid>
  );
};

export default Transaction;
