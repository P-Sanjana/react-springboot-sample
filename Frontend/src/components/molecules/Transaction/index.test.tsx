import React from "react";
import { render, screen } from "@testing-library/react";
import theme from "../../../theme/theme";
import Transaction from "./index";
import CheckLine from "../../../assets/icons/check-line.svg";

const tran = {
  date: "June 23",
  statusIcon: CheckLine,
  coin: "Bitcoin",
  symbol: "BTC",
  chipLabel: "Sold",
  coinAmount: "-0.0234510",
  price: "+$34,000.00",
  iconBgColor: theme.palette.success.main,
};

describe("Transaction component", () => {
  it("should have currency", () => {
    render(<Transaction {...tran} />);
    const currency = screen.getByTestId("currency");
    expect(currency).toBeInTheDocument();
  });
  it("should match snapshot", () => {
    const wrapper = render(<Transaction {...tran} />);
    expect(wrapper).toMatchSnapshot();
  });
});
