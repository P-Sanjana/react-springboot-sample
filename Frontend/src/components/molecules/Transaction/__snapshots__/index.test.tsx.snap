// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction component should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root makeStyles-container-14 MuiGrid-container MuiGrid-spacing-xs-2"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            class="MuiBox-root MuiBox-root-23"
          >
            <span
              class="MuiTypography-root makeStyles-date-16 MuiTypography-caption"
              data-testid="date"
            >
              June 23
            </span>
          </div>
          <div
            class="MuiBox-root MuiBox-root-24 makeStyles-rowContent-15"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-grid-xs-2"
            >
              <img
                class="makeStyles-statusIcon-18"
                data-testid="status"
                src="check-line.svg"
              />
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
            >
              <div
                class="MuiBox-root MuiBox-root-25"
              >
                <p
                  class="MuiTypography-root makeStyles-coin-17 MuiTypography-body1"
                  data-testid="currency"
                >
                  Bitcoin
                   
                  BTC
                </p>
                <div
                  class="MuiChip-root makeStyles-chip-19 MuiChip-sizeSmall"
                >
                  <span
                    class="MuiChip-label MuiChip-labelSmall"
                  >
                    Sold
                  </span>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
            >
              <div
                class="MuiBox-root MuiBox-root-26 makeStyles-columnContent-22"
                style="align-items: flex-end;"
              >
                <p
                  class="MuiTypography-root makeStyles-coinAmount-20 MuiTypography-body1"
                >
                  -0.0234510
                  BTC
                </p>
                <span
                  class="MuiTypography-root makeStyles-price-21 MuiTypography-caption"
                >
                  +$34,000.00
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root makeStyles-container-14 MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="MuiBox-root MuiBox-root-23"
        >
          <span
            class="MuiTypography-root makeStyles-date-16 MuiTypography-caption"
            data-testid="date"
          >
            June 23
          </span>
        </div>
        <div
          class="MuiBox-root MuiBox-root-24 makeStyles-rowContent-15"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-grid-xs-2"
          >
            <img
              class="makeStyles-statusIcon-18"
              data-testid="status"
              src="check-line.svg"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
          >
            <div
              class="MuiBox-root MuiBox-root-25"
            >
              <p
                class="MuiTypography-root makeStyles-coin-17 MuiTypography-body1"
                data-testid="currency"
              >
                Bitcoin
                 
                BTC
              </p>
              <div
                class="MuiChip-root makeStyles-chip-19 MuiChip-sizeSmall"
              >
                <span
                  class="MuiChip-label MuiChip-labelSmall"
                >
                  Sold
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
          >
            <div
              class="MuiBox-root MuiBox-root-26 makeStyles-columnContent-22"
              style="align-items: flex-end;"
            >
              <p
                class="MuiTypography-root makeStyles-coinAmount-20 MuiTypography-body1"
              >
                -0.0234510
                BTC
              </p>
              <span
                class="MuiTypography-root makeStyles-price-21 MuiTypography-caption"
              >
                +$34,000.00
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
